pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -260,6 +247,22 @@ func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 	}
 }
 
+// check if any resources in aks.cattle.io/v1
+func (e *aksOperatorController) checkCrdReady(cluster *mgmtv3.Cluster) (err error) {
+	resources, err := e.Discovery.ServerResourcesForGroupVersion("aks.cattle.io/v1")
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if errors.IsNotFound(err) || len(resources.APIResources) == 0 {
+		cluster, err = e.SetUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "Waiting on aks crd to be initialized")
```

Comment:
I think just return the error is sufficient and setting the message may not be necessary. It's nice but depending on if it is registered or not the expected condition is different here I believe. Also, if we were to update the entire object we probably want to return it from this function and reassign so that the cluster object in the parent function is up to date. This comment applies to all 3 occurrences of this function definition.

Reply:
Yeah, setting the condition in here so that if crd chart install failed, cluster could have a better way to display the information in UI showing the cluster is waiting on crd. For the cluster object it is passed as a pointer and reassigned in the method, it should get what's updated.

Reply:
Reassigning a pointer in a function doesn't bubble up to the caller: https://play.golang.org/p/rdD6A6alGIE

Reply:
Oh......

Reply:
@rmweir @thedadams This is fixed.